### PR TITLE
Template creator

### PR DESCRIPTION
**Pull request by david-i-berry**
Bug fix for GeoJSON and addition of additional command to create template for BUFR mapping.

Also, note change in argument/option from --mapping -> --bufr-template to be consistent with --geojson-template.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,46 +3,85 @@
 [![Tests](https://github.com/wmo-im/csv2bufr/workflows/tests%20%E2%9A%99%EF%B8%8F/badge.svg)](https://github.com/wmo-im/csv2bufr/actions/workflows/tests.yml)
 
 csv2bufr is a Python package to transform CSV data into WMO BUFR.  Currently, csv2bufr converts each row in a 
-CSV to a BUFR message.
+CSV to a BUFR message. Both a python module and CLI to the module are provided.
```
Comment (tomkralidis) [inline README.md:6]: `s/python/Python/`

File: README.md [hunk 2]
```diff
@@ -3,46 +3,85 @@
...
 
-### Usage
+### Installation
+The csv2bufr module relies on the ECCODES software library from ECMWF.
```
Comment (tomkralidis) [inline README.md:9]: ...relies on the [ecCodes](https://confluence.ecmwf.int/display/ECC) software library....

File: README.md [hunk 3]
```diff
@@ -3,46 +3,85 @@
...
-# now start the Docker container based on the image
-docker run -it -v ${pwd}:/app eccodes_v23
+### Configuration
 
-# export path to Python modules for this script
-export PYTHONPATH="${PYTHONPATH}:/app/src/csv2bufr/"
+Two configurations files are required, the first specifying how to map from the input CSV file to BUFR.
+The second contains the station level metadata from the WMO OSCAR/Surface catalogue. 
+Both files take the JSON format and a description of the mapping file is given below under usage.
+The second can be downloaded by the pyoscar tool (installed as part of the image build).
```
Comment (tomkralidis) [inline README.md:32]: ....can be downloaded by the [pyoscar](https://github.com/wmo-cop/pyoscar) tool ....

File: csv2bufr/__init__.py [hunk 4]
```diff
@@ -30,7 +30,7 @@
 import os.path
 from typing import Any, Iterator, Union
 from jsonpath_ng.ext import parser
-
+from copy import deepcopy
```
Comment (tomkralidis) [inline csv2bufr/__init__.py:33]: nitpick/your call: we should organize/alphabetize dependencies, in the following order:

- standard library
- installed packages
- local modules

Example:

```python
from copy import deepcopy
import csv
from datetime import timezone, datetime
import hashlib
from io import StringIO, BytesIO
import json
import logging
import os.path
from typing import Any, Iterator, Union

from eccodes import (codes_bufr_new_from_samples,
                     codes_set_array, codes_set, codes_get_native_type,
                     codes_write, codes_release, codes_get,
                     CODES_MISSING_LONG, CODES_MISSING_DOUBLE,
                     codes_bufr_keys_iterator_new,
                     codes_bufr_keys_iterator_next,
                     codes_bufr_keys_iterator_get_name, CodesInternalError)
from jsonpath_ng.ext import parser
from jsonschema import validate
```

File: csv2bufr/cli.py [hunk 5]
```diff
@@ -28,6 +28,7 @@
 
 from csv2bufr import __version__
```
Comment (tomkralidis) [inline csv2bufr/cli.py:29]:
```python
from csv2bufr import __version__, BUFRMessage, transform as transform_csv
```

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (david-i-berry) [reply README.md]: changed

File: README.md (hunk 2 above)
Comment (david-i-berry) [reply README.md]: changed.

File: README.md (hunk 3 above)
Comment (david-i-berry) [reply README.md]: done

File: csv2bufr/__init__.py (hunk 4 above)
Comment (david-i-berry) [reply csv2bufr/__init__.py]: changed.

File: csv2bufr/cli.py (hunk 5 above)
Comment (david-i-berry) [reply csv2bufr/cli.py]: changed.

**Review by tomkralidis — APPROVED** [review]
Thanks @david-i-berry.  Last comment is considering to Squash and Merge the commits, so the feature can be reverted if needed in one logical step.

**Pull request merged**